src/CacheMiddleware.php

```diff
@@ -377,6 +388,10 @@ private function invalidateCache(RequestInterface $request, ResponseInterface $r
     {
         $this->cacheStorage->delete($request);
 
+        foreach (array_diff(array_keys($this->httpMethods), [$request->getMethod()]) as $method) {
```

Comment:
You probably can propose a simpler implementation by removing the line 389 and iterate over all values from `$this->httpMethods`